myfitnesspal/entry.py

```diff
@@ -1,10 +1,24 @@
+import re
+
 from myfitnesspal.base import MFPBase
 
 
 class Entry(MFPBase):
     def __init__(self, name, nutrition):
         self._name = name
         self._nutrition = nutrition
+        
+        #split out quantity and measuring unit out of entry name
+        regex = r'(?P<short_name>.+), (?P<quantity>\d[\d\.]*) (?P<unit>[\w\(\)]+)(?: \(.*\))?'
```

Comment:
regex pattern changed . it now exhibits desired behaviour with all test cases

![image](https://user-images.githubusercontent.com/19669502/27171427-61574a96-51a9-11e7-91da-cb06ea43e091.png)